src/Middleware/HttpLogging/src/RequestBufferingStream.cs

```diff
@@ -138,6 +139,25 @@ public void LogRequestBody(HttpLoggingInterceptorContext logContext)
         _ => throw new NotImplementedException(_status.ToString()),
     };
 
+    private string GetStringInternal()
+    {
+        var result = _bodyBeforeClose ?? GetString(_encoding);
+        // Reset the value after its consumption to preserve GetString(encoding) behavior
+        _bodyBeforeClose = null;
+        return result;
+    }
+
+    public override void Close()
+    {
+        if (!HasLogged)
```

Comment:
Can we pass `HttpLoggingInterceptorContext` into the ctor like we do for the response stream and then check if body logging is enabled before getting the string?